src/init/features/hosting/index.ts

```diff
@@ -29,19 +33,124 @@ export async function doSetup(setup: any, config: any): Promise<void> {
   logger.info("have a build process for your assets, use your build's output directory.");
   logger.info();
 
+  let discoveredFramework = previews.frameworkawareness ?
+    await discover(config.projectDir, false) :
+    undefined;
+
+  if (previews.frameworkawareness) {
+
```

Comment:
Empty vertical whitespace doesn't help readability much - tabs indicate new blocks pretty efficiently, IMO. What do you think? Can the extra newlines be removed?